src/hotspot/share/opto/addnode.cpp

```diff
@@ -1263,6 +1263,20 @@ Node* MaxINode::Ideal(PhaseGVN* phase, bool can_reshape) {
   return IdealI(phase, can_reshape);
 }
 
+Node* MaxINode::Identity(PhaseGVN* phase) {
+  const TypeInt* t1 = phase->type(in(1))->is_int();
+  const TypeInt* t2 = phase->type(in(2))->is_int();
```

Comment:
I think we can't encounter `TOP` here because we run `Value()` before `Identity()`, so if TOP is returned by Value() the idealization process exits to return the top node before running Identity().